packages/ui/src/icons/arrow-down-icon.native.tsx

```diff
@@ -0,0 +1,20 @@
+import { Component, forwardRef } from 'react';
+
+import { IconVariantMap } from 'src/icons/types.shared';
+
+import ArrowDownSmall from '../assets/icons/arrow-down-16-16.svg';
+import ArrowDown from '../assets/icons/arrow-down-24-24.svg';
+import { Icon, IconProps } from './icon/icon.native';
+
+const variants: IconVariantMap = {
+  small: <ArrowDownSmall />,
+  default: <ArrowDown />,
+};
+
+export const ArrowDownIcon = forwardRef<Component, IconProps>(({ variant, ...props }, ref) => {
+  return (
+    <Icon ref={ref} {...props}>
+      {variants[variant ?? 'default']}
```

Comment:
Is it nec to make this change when all other icons use a different code pattern?

Reply:
No, was just playing with patterns and thinking about suggesting a different way of doing this down the line.

Actually as I was looking at this I was wondering why we don't just have a single component that can render any icon, eg `<LeatherIcon icon="icon-name" variant="small" {...props} />` - I find that pattern makes adding icons much easier as you just have a lookup table / function and all new icons can just be added there. 

Was the idea with this implementation that it allows for better tree-shaking? 

I can revert it to how the others do it for now.

Reply:
eg [FontAwesome](https://docs.fontawesome.com/v5/web/use-with/react)

Reply:
Most of our components are based off Radix. https://www.radix-ui.com/icons

Reply:
Was thinking on this, might be best to get @tigranpetrossian opinion here bc he is mostly owning the UI library? I don't have a strong opinion if others prefer a diff setup. It would just be quite a chunk of work to refactor as they are currently used this way in both web and mobile.